File: service/src/main/java/net/huray/backend/minuting/service/MinutesReadService.java
package net.huray.backend.minuting.service;

import lombok.RequiredArgsConstructor;
import net.huray.backend.minuting.dto.MinutesDto;
import net.huray.backend.minuting.entity.MinutesEntity;
import net.huray.backend.minuting.repository.MinutesRepository;
import org.modelmapper.ModelMapper;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.server.ResponseStatusException;

import java.util.List;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class MinutesReadService {

    private final MinutesRepository minutesRepository;
    private final ModelMapper modelMapper;

    public List<MinutesDto.MinutesSimple> list() {
        return minutesRepository.findAll().stream()
                .map(it -> modelMapper.map(it, MinutesDto.MinutesSimple.class))
                .toList();
    }

    public MinutesDto.MinutesDetail getDetailById(Long id) {
        MinutesEntity minutes = minutesRepository.findById(id)
                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));

        return modelMapper.map(minutes, MinutesDto.MinutesDetail.class);
    }
}

File: service/src/main/kotlin/net/huray/backend/minuting/service/MinutesReadService.kt
package net.huray.backend.minuting.service

import net.huray.backend.http.exception.BaseException
import net.huray.backend.http.exception.code.ErrorCode
import net.huray.backend.minuting.dto.MinutesDto
import net.huray.backend.minuting.repository.MinutesRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class MinutesReadService(
    private val minutesRepository: MinutesRepository
) {

    fun list() = minutesRepository.findAll()
        .map { MinutesDto.MinutesSimple(it.id, it.title, it.contents) }

    fun getDetailById(id: Long) = minutesRepository.findById(id)
        .orElseThrow { throw BaseException(ErrorCode.MINUTES_NOT_FOUND, id) }
        .let {
            MinutesDto.MinutesDetail(it.id, it.title, it.contents).apply {
                this.createdAt = it.createdAt
                this.updatedAt = it.updatedAt
            }
        }

}